### ISSUE TITLE
How to pre-train each object field separately?

### ISSUE THREAD

**Issue by Szy-Young** [andrewsonga/Total-Recon#1]
Congrats to your great work and thank you very much for releasing the code!

I have a question about the pre-training stage:
According to the Equations (5)-(7), volume rendering of the silhouette for a specific j-th object will also involve the other object fields. This seems to ensure the rendered object silhouette to be occlusion-aware. So in the pre-training stage, how to optimize each object field separately? Does that mean we can temporarily ignore the occlusion between objects during pre-training and leave it for the finetuning stage? But I see the silhouette loss is removed in fine-tuning, so this makes me a bit confused. Could you help me make it clear?

Thanks in advance for taking a look & thank you for your help!
Best regards.

**Comment by andrewsonga** [comment]
Hello Ziyang,

Thank you for the kind words about our project.
These are great questions that you are asking!

You are correct that equations 5~7 ensure that the our method is occlusion-aware during joint-finetuning. You are also correct that during the pre-training of each object field, we do temporarily ignore the occlusion between objects; we pretrain each object field with modal object masks (as opposed to amodal masks), and we observed that this doesn’t hurt the object’s reconstruction quality too much as the deformation field can still account for the occluded portions of the object.

By the time we start joint-finetuning, the object fields have reasonably converged so we now remove the (potentially noisy) silhouette loss and have the composite model polish object boundaries in a more self-supervised manner. We do this by solely relying on the reconstruction losses for color, depth, and 2d optical flow as well as per-object 3D cycle-consistency losses to regularize the deformation fields.

We will release the pretraining and joint-finetuning code very soon. Please stay tuned for updates :)

**Comment by Szy-Young** [comment]
@andrewsonga Many thanks for the detailed answer! My confusions are all clear now
